units.lisp

```diff
@@ -0,0 +1,262 @@
+     (define-unit-prefix |tebi| 4 :abbrev |Ti| :base 1024)
+     (define-unit-prefix |pebi| 5 :abbrev |Pi| :base 1024)
+     (define-unit-prefix |exbi| 6 :abbrev |Ei| :base 1024)
+     (define-unit-prefix |zebi| 7 :abbrev |Zi| :base 1024)
+     (define-unit-prefix |yobi| 8 :abbrev |Yi| :base 1024)
+
+     ;; Most units are named according to their name in CSS
+     (define-unit |m| :alias (|meter| |metres| |meters|) :prefix-test (pq::prefix-range 10 nil 3))
+     (define-unit |s| :alias (|second| |seconds|) :prefix-test (pq::prefix-range 10 nil -3))
+     (define-unit |rad| :def (1) :alias (|radian| |radians|) :prefix-test (pq::prefix-range 10 nil -3))
+     (define-unit |grad| :def (400/360 |rad|) :alias (|gradian| |gradians|) :prefix-test (pq::prefix-range 10 nil -3))
+     (define-unit |turn| :def ((* 2 pi) |rad|) :alias |turns| :prefix-test (constantly nil))
+     (define-unit |steradian| :def (1) :abbrev |sr| :prefix-test (pq::prefix-range 10 nil -3))
+     (define-unit |Hz| :def (1 / |second|) :alias |hertz|  :prefix-test (pq::prefix-base 10 3))
+     (define-unit |byte| :def (1) :alias |bytes| :abbrev |b| :prefix-test (pq::prefix-or (pq::prefix-base 1024) (pq::prefix-range 10 3 nil)))
+     (define-unit |minute| :def (60 |s|) :alias |minutes| :abbrev |min| :prefix-test (constantly nil))
+     (define-unit |hour| :def (60 |minute|) :alias |hours| :abbrev |h| :prefix-test (constantly nil))
+     (define-unit |day| :def (24 |hour|) :alias |days| :abbrev |d| :prefix-test (constantly nil))
+     (define-unit |deg| :def ((/ pi 180) |rad|) :alias (|degrees| |degree|) :prefix-test (pq::prefix-range 10 nil -3))
+     (define-unit |%| :def (0.01) :alias (|percent| |percents|) :prefix-test (constantly nil))
+     (define-unit |in| :def (0.0254 |m|) :alias (|inch| |inches|) :prefix-test (constantly nil))
+     (define-unit |pt| :def (1/72 |inch|) :alias (|point| |points|) :prefix-test (constantly nil))
+     (define-unit |pc| :def (1/6 |inch|) :alias (|pica| |picas|) :prefix-test (constantly nil))
+     (define-unit |px| :def (1/96 |inch|) :alias (|pixel| |pixels|) :prefix-test (constantly nil))
+     ;; assuming the most common font size, used only in mixed calculations
+     (define-unit |em| :def (16 |px|) :prefix-test (constantly nil))
+
+     ,@body))
+
+(defun parse-css-number (x &key unit)
+  "Parses CSS number wrapping it in pq:quantity if unit of measure is present"
+  (cond ((stringp x)
+         (multiple-value-bind (val idx) (parse-float x :junk-allowed t)
+           (if unit
+               (make-quantity :value val :unit (make-unit (list unit 1)))
+               (if (>= idx (length x))
+                   val
+                   (make-quantity :value val
+                                  :unit (make-unit
+                                         (list (subseq x
+                                                       ;;; TODO remove this hack after parse-float is fixed
+                                                       (if (or
```

Comment:
Instead of this `or` you can use `(char-equal #\e ..)`

Reply:
Good one